### PR TITLE
ddl: fix data race in schema_test

### PR DESCRIPTION
**Pull request by zimulala**
## What have you changed? (mandatory)
Fix data race in schema_test.
```
WARNING: DATA RACE
Read at 0x00c420af55f0 by goroutine 29:
  github.com/pingcap/tidb/ddl.(*ddl).asyncNotifyWorker()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/ddl.go:457 +0xf6
  github.com/pingcap/tidb/ddl.(*ddl).doDDLJob()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/ddl.go:475 +0x174
  github.com/pingcap/tidb/ddl.testRunInterruptedJob.func1()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/schema_test.go:221 +0x5c

Previous write at 0x00c420af55f0 by goroutine 124:
  github.com/pingcap/tidb/ddl.(*ddl).start()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/ddl.go:359 +0x28a
  github.com/pingcap/tidb/ddl.testRunInterruptedJob()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/schema_test.go:232 +0x20d
 2018/07/31 10:47:04.655 ddl_worker.go:109: [info] [ddl] start DDL worker 0, tp general
  github.com/pingcap/tidb/ddl.(*testTableSuite).TestTableResume()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/table_test.go:267 +0x4d1
```
```
WARNING: DATA RACE
Read at 0x00c420c50720 by goroutine 29:
  runtime.mapaccess1()
      /usr/local/go/src/runtime/hashmap.go:343 +0x0
  github.com/pingcap/tidb/ddl.(*ddl).asyncNotifyWorker()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/ddl.go:457 +0x120
  github.com/pingcap/tidb/ddl.(*ddl).doDDLJob()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/ddl.go:475 +0x174
  github.com/pingcap/tidb/ddl.testRunInterruptedJob.func1()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/schema_test.go:221 +0x5c

Previous write at 0x00c420c50720 by goroutine 124:
  runtime.mapassign()
      /usr/local/go/src/runtime/hashmap.go:505 +0x0
  github.com/pingcap/tidb/ddl.(*ddl).start()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/ddl.go:360 +0x355
  github.com/pingcap/tidb/ddl.testRunInterruptedJob()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/schema_test.go:232 +0x20d
  github.com/pingcap/tidb/ddl.(*testTableSuite).TestTableResume()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/table_test.go:267 +0x4d1
```

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
Yes.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No.
## Does this PR affect tidb-ansible update? (mandatory)
No.
## Does this PR need to be added to the release notes? (mandatory)
No.

### CODE REVIEW

File: ddl/ddl_test.go [hunk 1]
```diff
@@ -61,6 +62,24 @@ func (d *ddl) generalWorker() *worker {
 	return d.workers[generalWorker]
 }
 
+// restartWorkers is like the function fo d.start. But it won't initialize the "workers" and create a new worker.
```
Comment (crazycs520) [inline ddl/ddl_test.go]: s/fo/for?

Comment (zhexuany) [reply ddl/ddl_test.go]: What do you mean by "is like"?

Comment (zimulala) [reply ddl/ddl_test.go]: The operation of these two functions is similar.

**Comment by crazycs520** [comment]
LGTM

**Comment by zimulala** [comment]
PTAL @crazycs520 @zhexuany  @winkyao

File: ddl/ddl_test.go (hunk 1 above)
Comment (zhexuany) [reply ddl/ddl_test.go]: For this case, either is like the function d.start() or is similar with the function d.start()

**Pull request merged**